### PR TITLE
HPCC-24939 Spurious tracing output by one-shot roxie workunits

### PR DESCRIPTION
**Pull request by richardkchapman**
Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [x] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [x] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [x] Performance
  - [x] Security
  - [x] Thread-safety
  - [x] Cloud-compatibility
  - [x] Premature optimization
  - [x] Existing deployed queries will not be broken
  - [x] This change fixes the problem, not just the symptom
  - [x] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

## Testing:
Tested manually under Docker Desktop, observing roxie logs.

### CODE REVIEW

**Comment by richardkchapman** [comment]
@ghalliday Please review

File: roxie/ccd/ccdfile.cpp [hunk 1]
```diff
@@ -1425,7 +1427,10 @@ class CRoxieFileCache : implements IRoxieFileCache, implements ICopyFileProgress
             toClose.interrupt();
             bct.join(timeout);
             hct.join(timeout);
+        }
 #ifdef _CONTAINERIZED
+        if (cidtActive)
```
Comment (ghalliday) [inline roxie/ccd/ccdfile.cpp:1432]: should there be a similar test in wait() below?

**Comment by ghalliday** [comment]
@richardkchapman build is failing

File: roxie/ccd/ccdfile.cpp (hunk 1 above)
Comment (richardkchapman) [reply roxie/ccd/ccdfile.cpp]: Probably - I will add

**Comment by richardkchapman** [comment]
@ghalliday I fixed the build error in non-containerized, and addressed your other comment. Please re-review

File: roxie/ccd/ccdqueue.cpp [hunk 2]
```diff
@@ -2615,50 +2615,59 @@ class PacketDiscarder : public Thread, implements IPacketDiscarder
...
-            else if (!anyActivity)
-            {
-                // to avoid leaking partial unwanted packets, we clear out mc periodically...
-                ROQ->queryReceiveManager()->detachCollator(mc);
-                mc.setown(ROQ->queryReceiveManager()->createMessageCollator(rowManager, RUID_DISCARD));
             }
         }
+        catch (IException * E)
+        {
+            if (!aborted || QUERYINTERFACE(E, InterruptedSemaphoreException) == NULL)
```
Comment (ghalliday) [inline roxie/ccd/ccdqueue.cpp:2667]: trivial: We should use dynamic_cast instead on new code.  Possibly replace all QUERYINTERFACE calls at some point.

**Review by ghalliday — COMMENTED** [review]
@richardkchapman looks good.

**Pull request merged**